planetiler-core/src/main/java/com/onthegomap/planetiler/FeatureMerge.java

```diff
@@ -502,6 +506,41 @@ private static void extractPolygons(Geometry geom, List<Polygon> result, double
     }
   }
 
+  private static double isFill(double outerArea, LinearRing exteriorRing) {
+    if (outerArea < 256 * 256) {
+      return -1;
+    }
+    double proposedBuffer = (Math.sqrt(outerArea) - 256) / 2;
+    double tol = proposedBuffer / 10;
+    double min = -proposedBuffer;
+    double max = 256 + proposedBuffer;
+    var cs = exteriorRing.getCoordinateSequence();
+    for (int i = 0; i < cs.size(); i++) {
+      double x = cs.getX(i);
+      double y = cs.getY(i);
+      if ((notClose(x, min, tol) && notClose(x, max, tol)) && (notClose(y, min, tol) && notClose(y, max, tol))) {
```

Comment:
this just checks that all vertices are on the fill edge... I _think_ that covers all possible cases here, but might want something more robust like this: https://github.com/onthegomap/planetiler/blob/de4f99c4a3156389cead6016d17c2ef12ebc80b9/planetiler-core/src/main/java/com/onthegomap/planetiler/VectorTile.java#L854 that checks whether any segment crosses the inside?